modules/mobile/app/controllers/mobile/v0/users_controller.rb

```diff
@@ -42,13 +42,19 @@ def map_logingov_to_idme
         end
       end
 
-      def link_user_with_vet360
-        Mobile::V0::Vet360LinkingJob.perform_async(@current_user.uuid)
-      end
-
       def user_accessible_services
         @user_accessible_services ||= Mobile::V0::UserAccessibleServices.new(current_user)
       end
+
+      def handle_vet360_id
+        if @current_user.vet360_id.blank?
+          Mobile::V0::Vet360LinkingJob.perform_async(@current_user.uuid)
+        elsif (mobile_user = Mobile::User.find_by(icn: @current_user.icn))
+          Rails.logger.info('Mobile Vet360 account linking was successful request succeeded for user with uuid',
+                            { user_icn: @current_user.icn, attempts: mobile_user.vet360_linking_attempts })
+          mobile_user.destroy
```

Comment:
You still need to update this to match, but I assume this is probably still a WIP.